cvat/apps/dataset_manager/tests/test_rest_api_formats.py

```diff
@@ -182,12 +288,794 @@ def _generate_url_dump_tasks_annotations(self, task_id):
+                        "WiderFace 1.0", "VGGFace2 1.0", \
+                    ]:
+                        self._create_annotations(task, dump_format_name, "default")
+                    else:
+                        self._create_annotations(task, dump_format_name, "random")
+
+                    task_ann = TaskAnnotation(task_id)
+                    task_ann.init_from_db()
+                    task_ann_prev_data = task_ann.data
+                    url = self._generate_url_dump_tasks_annotations(task_id)
+
+                    file_zip_name = osp.join(test_dir, f'{test_name}_{dump_format_name}.zip')
+                    data = {
+                        "format": dump_format_name,
+                        "action": "download",
+                    }
+                    self._download_file(url, data, self.admin, file_zip_name)
+                    self.assertEqual(osp.exists(file_zip_name), True)
+
+                    self._remove_annotations(url, self.admin)
+
+                    self._create_annotations(task, "CVAT for images 1.1 many jobs", "default")
+
+                    if dump_format_name == "CVAT for images 1.1" or dump_format_name == "CVAT for video 1.1":
+                        dump_format_name = "CVAT 1.1"
+                    url = self._generate_url_upload_tasks_annotations(task_id, dump_format_name)
+
+                    with open(file_zip_name, 'rb') as binary_file:
+                        self._upload_file(url, binary_file, self.admin)
+                    task_ann = TaskAnnotation(task_id)
+                    task_ann.init_from_db()
+                    task_ann_data = task_ann.data
+                    self.assertEqual(len(task_ann_data["shapes"]), len(task_ann_prev_data["shapes"]))
+
+    def test_api_v1_unit_test_on_normalize_shape_function(self):
+        # 3 points
+        TrackManager.normalize_shape({
+             "points": [1.5, 2.5],
+        })
+
+        # 4 points
+        TrackManager.normalize_shape({
+            "points": [1.5, 2.5, 0.5, 8.6, 9.6, 3.6, 2.8, 7.2],
+        })
+        # 1 point
+        TrackManager.normalize_shape({
+            "points": [1.5, 2.5],
+        })
+        # empty shape
+        with self.assertRaises(ValueError):
+            TrackManager.normalize_shape({
+                "points": [],
+            })
+        # invalid count of points
+        with self.assertRaises(ValueError):
+            TrackManager.normalize_shape({
+                "points": [1.5, 2.5, 7.5],
+            })
+        # negative points
+        TrackManager.normalize_shape({
+            "points": [-1.5, 2.5, -9.8, -4.6],
+        })
+        # integer
+        TrackManager.normalize_shape({
+            "points": [1, 2, 9, 4],
+        })
+
+    def test_api_v1_tasks_annotations_dump_and_upload_many_jobs_with_datumaro(self):
+        test_name = self._testMethodName
+        upload_format_name = "CVAT 1.1"
+
+        for include_images in (False, True):
```

Comment:
Use `itertools.product(iter1, iter2, ...)` to make test parameter matrices. Other ways https://stackoverflow.com/questions/32899/how-do-you-generate-dynamic-parameterized-unit-tests-in-python

Reply:
Done.